src/validation/obfl.rng

```diff
@@ -792,24 +792,38 @@
     <element name="table-of-contents">
       <ref name="attlist-table-of-contents"/>
       <oneOrMore>
-        <ref name="toc-entry"/>
+        <ref name="toc-block"/>
       </oneOrMore>
     </element>
   </define>
   <define name="attlist-table-of-contents" combine="interleave">
     <ref name="name-id"/>
   </define>
-  <define name="toc-entry">
-    <element name="toc-entry">
-      <ref name="attlist-toc-entry"/>
+  <define name="toc-block">
+    <element name="toc-block">
+      <ref name="attlist-toc-block"/>
       <zeroOrMore>
         <choice>
-          <ref name="blockContents"/>
           <ref name="toc-entry"/>
+          <ref name="toc-block"/>
         </choice>
       </zeroOrMore>
     </element>
   </define>
+  <define name="attlist-toc-block" combine="interleave">
+    <ref name="border-atts"/>
+    <ref name="margin-atts"/>
+    <ref name="padding-atts"/>
+    <ref name="block-atts"/>
+  </define>
+  <define name="toc-entry">
+    <element name="toc-entry">
+      <ref name="attlist-toc-entry"/>
+      <zeroOrMore>
+        <ref name="blockContents"/>
+      </zeroOrMore>
+    </element>
+  </define>
   <define name="attlist-toc-entry" combine="interleave">
```

Comment:
`blockAtts` should be removed from `attlist-toc-entry`

Reply:
I made changes so that `toc-entry` has only this attribute: `ref-id`.